### PR TITLE
Improves Microsoft.Bot.Builder.Tests

### PR DESCRIPTION
**Pull request by iscaltritti**
## Description

Tests within _Microsoft.Bot.Builder.Tests_ featured many use instances of _Task.Delay_ that either aren't necessary or can been shortened without impacting the tests effectiveness, improving the tests time consumption.

The removals and reductions add up to **6.05 seconds saved**.

## Specific Changes

  - _Task.Delay_ removed from _BotFrameworkAdapterOAuthTests_, _TelemetryMiddlewareTests_ and _Transcript_MiddlewareTests_.
  - _Task.Delay_ time waited reduced in _TestFlowTests_ and _ShowTyping_MiddlewareTests_. Time chosen based on brute force testing.

## Testing

The following image showcases a local test comparison between the current implementation and the improvement.

image

### CODE REVIEW

**Comment by denscollo** [comment]
This PR doesn't addresses an issue, since # 4442 was only a issue to review and create a report. We don't have an issue yet for this, so please remove it

File: tests/Microsoft.Bot.Builder.Tests/ShowTyping_MiddlewareTests.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public async Task ShowTyping_TestMiddleware_1_Second_Interval()
 
             await new TestFlow(adapter, async (context, cancellationToken) =>
             {
-                await Task.Delay(TimeSpan.FromMilliseconds(2800));
+                await Task.Delay(TimeSpan.FromMilliseconds(2300));
```
Comment (denscollo) [inline tests/Microsoft.Bot.Builder.Tests/ShowTyping_MiddlewareTests.cs:22]: Why from 2800 to 2300? Was this arbitrary? On the description says 

> time waited reduced by half

Comment (iscaltritti-sw) [reply tests/Microsoft.Bot.Builder.Tests/ShowTyping_MiddlewareTests.cs:22]: I overlooked this value while writing the original PR, thanks for pointing it out.
I corrected the description and clarified that both values were chosen based on brute force testing.

**Comment by iscaltritti-sw** [comment]
Corrected the PR's description based on feedback:

- Removed line `Addresses Issue # 4442`.
- Grammar improvement.
- Corrected and explained the reasoning behind the times chosen for _Task.Delay_ reductions.

**Comment by iscaltritti-sw** [comment]
- Updated comparison image to dotnet 3.1 instead of 2.1.

**Pull request closed without merging**